packages/mcdoc/src/binder/index.ts

```diff
@@ -1215,9 +1215,9 @@ function convertLiteralValue(
 	} else if (TypedNumberNode.is(node)) {
 		const { suffix, value } = TypedNumberNode.destruct(node)
 		return {
-			kind: 'number',
+			kind: convertLiteralNumberSuffix(suffix, ctx) ??
+				(Number.isInteger(value.value) ? 'int' : 'double'),
```

Comment:
`Number.isInteger(1.0)` would be `true` so we would have an `int` literal number when it is in fact a floating point number. Probably have to change the spec and the parser to properly fix this (the spec current specifies `float` + suffix instead of using `float` and `interger` properly cuz i was too lazy).

Reply:
Hmm, good point

Reply:
I changed the parser to emit an integer node when there is no dot and no float type marker